lib/data_services_api/dataset.rb

```diff
@@ -49,15 +49,54 @@ def structure
     end
 
     def query(query)
-      service.api_post_json(data_api, query.to_json)
+      converter = SapiNTConverter.new(query.to_json)
+      sapi_query_params = converter.to_sapint_query
+      sapint_response = service.api_get_json(data_api, sapi_query_params)
+      to_dsapi_response(sapint_response)
     end
 
     def describe(uri)
       service.api_get_json(describe_api, uri: uri)
     end
 
     def explain(query)
-      service.api_post_json(explain_api, query.to_json)
+      converter = SapiNTConverter.new(query.to_json)
+      sapi_query_params = converter.to_sapint_query
+      explain_url = "#{data_api}/explain"
+      { sparql: service.api_get_text(explain_url, sapi_query_params) }
+    end
+
+    private
+
+    # Converts SAPINT returned JSON format to DSAPI returned JSON format
+    def to_dsapi_response(sapint_response)
```

Comment:
Converting to the DsAPI syntactic form is not a responsibility of a `Dataset` per se. I think it should be a separate class. That will also make it easier to write the tests